app/adapters/signal_adapter/inbound.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module SignalAdapter
+  UNKNOWN_CONTRIBUTOR = :unknown_contributor
+  UNKNOWN_CONTENT = :unknown_content
+
+  class Inbound
+    UNKNOWN_CONTENT_KEYS = %w[mentions attachments contacts].freeze
+
+    attr_reader :sender, :text, :message
+
+    def initialize
+      @callbacks = {}
+    end
+
+    def on(callback, &block)
+      @callbacks[callback] = block
+    end
+
+    def consume(signal_message)
+      signal_message = signal_message.with_indifferent_access
+
+      @text = signal_message.dig(:envelope, :dataMessage, :message)
+      @sender = initialize_sender(signal_message)
+      return unless @sender
+
+      @message = initialize_message(signal_message)
+      return unless @message
+
+      yield(@message) if block_given?
+    end
+
+    private
+
+    def trigger(event, *args)
+      return unless @callbacks.key?(event)
+
+      @callbacks[event].call(*args)
+    end
+
+    def initialize_sender(signal_message)
+      phone_number = signal_message.dig(:envelope, :source)
+      sender = Contributor.find_by(phone_number: phone_number)
```

Comment:
Let's rename this to `signal_phone_number` in order to keep the naming of channel-specific attributes consistent

Reply:
You have to distinguish between the one `signal_phone_number` of the server and many `signal_phone_number`s of each contributor. Suggestions?

Reply:
What about `server_signal_phone_number` for disambiguation?

Reply:
changed it to `signal_server_phone_number` for a consistent naming scheme